units/crons.go

```diff
@@ -448,95 +448,95 @@ func PopulateSchedulerJobs(env evergreen.Environment) amboy.QueueOperation {
-				"message": "CheckBlockedTasks job is disabled",
+				"message": "scheduler is disabled",
 				"impact":  "new tasks are not enqueued",
 				"mode":    "degraded",
 			})
 			return nil
 		}
-
 		config, err := evergreen.GetConfig()
 		if err != nil {
 			return errors.WithStack(err)
 		}
-
 		catcher := grip.NewBasicCatcher()
+
+		lastPlanned, err := model.FindTaskAliasQueueLastGenerationTimes()
+		catcher.Add(err)
+
 		// find all active distros
 		distros, err := distro.Find(distro.ByNeedsPlanning(config.ContainerPools.Pools))
 		catcher.Add(err)
 
+		lastRuntime, err := model.FindTaskQueueGenerationRuntime()
+		catcher.Add(err)
+
 		ts := utility.RoundPartOfMinute(0)
 		for _, d := range distros {
-			catcher.Add(queue.Put(ctx, NewCheckBlockedTasksJob(d.Id, ts)))
+			// do not create scheduler jobs for parent distros
+			if d.IsParent(config) {
+				continue
+			}
+
+			lastRun, ok := lastPlanned[d.Id]
+			if ok && time.Since(lastRun) < 2*time.Minute {
+				continue
+			}
+			if ok && time.Since(lastRun)+10*time.Second < lastRuntime[d.Id] {
+				continue
+			}
+
+			catcher.Wrapf(amboy.EnqueueUniqueJob(ctx, queue, NewDistroAliasSchedulerJob(d.Id, ts)), "distro 's'", d.Id)
 		}
-		return catcher.Resolve()
+
+		return errors.Wrap(catcher.Resolve(), "populating distro alias scheduler jobs")
 	}
 }
 
-func PopulateAliasSchedulerJobs(env evergreen.Environment) amboy.QueueOperation {
+func PopulateCheckUnmarkedBlockedTasks() amboy.QueueOperation {
 	return func(ctx context.Context, queue amboy.Queue) error {
 		flags, err := evergreen.GetServiceFlags()
 		if err != nil {
 			return errors.WithStack(err)
 		}
-
-		if flags.SchedulerDisabled {
+		if flags.CheckBlockedTasksDisabled {
 			grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
-				"message": "scheduler is disabled",
+				"message": "CheckBlockedTasks job is disabled",
 				"impact":  "new tasks are not enqueued",
 				"mode":    "degraded",
 			})
 			return nil
 		}
+
 		config, err := evergreen.GetConfig()
 		if err != nil {
 			return errors.WithStack(err)
 		}
-		catcher := grip.NewBasicCatcher()
-
-		lastPlanned, err := model.FindTaskAliasQueueLastGenerationTimes()
-		catcher.Add(err)
 
+		catcher := grip.NewBasicCatcher()
 		// find all active distros
 		distros, err := distro.Find(distro.ByNeedsPlanning(config.ContainerPools.Pools))
 		catcher.Add(err)
 
-		lastRuntime, err := model.FindTaskQueueGenerationRuntime()
-		catcher.Add(err)
-
 		ts := utility.RoundPartOfMinute(0)
 		for _, d := range distros {
-			// do not create scheduler jobs for parent distros
-			if d.IsParent(config) {
-				continue
-			}
-
-			lastRun, ok := lastPlanned[d.Id]
-			if ok && time.Since(lastRun) < 2*time.Minute {
-				continue
-			}
-			if ok && time.Since(lastRun)+10*time.Second < lastRuntime[d.Id] {
-				continue
-			}
-
-			catcher.Add(queue.Put(ctx, NewDistroAliasSchedulerJob(d.Id, ts)))
+			catcher.Add(queue.Put(ctx, NewCheckBlockedTasksJob(d.Id, ts)))
```

Comment:
Should this be `EnqueueUniqueJob`?

Reply:
I didn't change this job, nor do I plan to change it. The diff is lying and this line hasn't changed at all - I moved the distro/alias scheduler functions to be closer together.